software/util/sdojsonldcontext.py

```diff
@@ -16,99 +15,68 @@
 import software
 import software.SchemaTerms.sdotermsource as sdotermsource
 import software.SchemaTerms.sdoterm as sdoterm
+import software.util.pretty_logger as pretty_logger
 
 log = logging.getLogger(__name__)
 
 
-def createcontext():
-    """Generates a basic JSON-LD context file for schema.org.
+CONTEXT = None
+SCHEMAURI = "http://schema.org/"
+
 
-    TODO(wiesmann): this function is currenty buggy:
-    * it outputs invalid JSON if `getAllTerms` returns an empty set.
-    * if a property has multiple types, the @type has multiple values,
-      which is invalid JSON.
+def getContext():
+    global CONTEXT
+    if not CONTEXT:
+        CONTEXT = createcontext()
+    return CONTEXT
 
-    """
-    log.info("Creating JSON-LD context")
 
-    SCHEMAURI = "http://schema.org/"
+def _convertTypes(type_range):
```

Comment:
can you add a type hint to `type_range` it would help understand this function.

Reply:
Done